### PR TITLE
use single Python Version for API container CI/CD and tox

### LINKED ISSUE #717: build: use single Python Version for API container CI/CD and tox
### What is the task about?

RSTUF API is provided as a Container Image.

Currently, we support during the tests (tox) and CI/CD process (Github Actions) multiple versions.
It is not required, as we build the image using Python version 3.12.

We can remove all the previous versions and operate with one specific version (3.12).

Se the reference below.

### Parent feature

_No response_

### References

https://github.com/repository-service-tuf/repository-service-tuf-worker/pull/629/commits/8f8412521a6be89b6ff3eb8c1012d23f96564d81

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Brijeshthummar02**
# Description
use single Python Version for API container CI/CD and tox 

Fixes #717 

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

**Comment by Brijeshthummar02** [comment]
@kairoaraujo please go through the pr.

**Review by kairoaraujo — CHANGES_REQUESTED** [review]
Missing changes on[ tox.ini](https://github.com/repository-service-tuf/repository-service-tuf-api/blob/main/tox.ini)

**Review by MVrachev — COMMENTED** [review]
Could you also change those places to python3.12:
- https://github.com/repository-service-tuf/repository-service-tuf-api/blob/ba2190f647c86456b4f1b41236c383e6a3c604e7/docker-compose.yml#L73
- https://github.com/repository-service-tuf/repository-service-tuf-api/blob/ba2190f647c86456b4f1b41236c383e6a3c604e7/README.rst#development-tools - to `Python >=3.12`
- https://github.com/repository-service-tuf/repository-service-tuf-api/blob/ba2190f647c86456b4f1b41236c383e6a3c604e7/.github/workflows/publish_container.yml#L33
- https://github.com/repository-service-tuf/repository-service-tuf-api/blob/ba2190f647c86456b4f1b41236c383e6a3c604e7/.github/workflows/update-pre-commit-hooks.yml#L14
- https://github.com/repository-service-tuf/repository-service-tuf-api/blob/ba2190f647c86456b4f1b41236c383e6a3c604e7/.github/workflows/ci.yml#L15   @kairoaraujo I think we don't need to run it for  `3.10` and `3.11`, correct? We can only use `3.12`?

**Comment by Brijeshthummar02** [comment]
@MVrachev there you go, i have made changes as you mentioned.

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ services:
         condition: service_healthy
 
   web:
-    image: python:3.10-slim-buster
+    image: python:3.12-slim-buster
```
Comment (kairoaraujo) [inline docker-compose.yml:73]:
```suggestion
    image: python:3.12-slim
```
There is no buster for 3.12 slim

**Review by kairoaraujo — CHANGES_REQUESTED** [review]

**Comment by Brijeshthummar02** [comment]
@kairoaraujo resolved as mentioned by you.

**Review by kairoaraujo — APPROVED** [review]

**Comment by Brijeshthummar02** [comment]
@kairoaraujo is it good to go now?

**Comment by kairoaraujo** [comment]
Thank you @Brijeshthummar02 
We hope you engage in the RSTUF project.

**Comment by Brijeshthummar02** [comment]
@kairoaraujo yeah, i would stick to the RSTFU projects.

**Pull request merged**